tests/test_cases/mutex.c

```diff
@@ -0,0 +1,30 @@
+#include<pthread.h>
+#include<stdlib.h>
+#include<sys/wait.h>
+#include<stdio.h>
+#include<unistd.h>
+pthread_mutex_t pmt;
+pthread_cond_t pct;
+
+void* lock_and_unlock(void* _) {
+        pthread_mutex_lock(&pmt);
+        printf("locked child\n");
+        pthread_mutex_unlock(&pmt);
+        pthread_cond_signal(&pct);
+}
```

Comment:
The indentation should be standardized (my fault but should be fixed before merge